### DISCUSSION TITLE
How to use store with initialState value inside Provider in Next.js with Jotai v2?

### DISCUSSION THREAD

**Discussion by AlexTechNoir** [pmndrs/jotai#1867] [Q&A] [answered]
Hello!

I'm migrating Jotai v1 -> v2 in Next.js and I'm using `initialState` value in `Provider` inside `initialValues` prop:

`_app.js`:
```javascript
import { Provider } from 'jotai'
import { someAtom, someOtherAtom } from '../store'

export default function MyApp({ Component, pageProps }) {
  const { initialState } = pageProps
  
  return(
    <Provider initialValues={ initialState && [[ someAtom, someOtherAtom, initialState ]]}>
      <Component {...pageProps} />
    </Provider>
  )
}
```
Docs say I should use `store` instead of `initialValues`. How can I adapt it in case with `initialState` value? 

Examples with Next.js in CodeSandbox are using Jotai v1

**Comment by dai-shi** [comment, marked as answer]
https://jotai.org/docs/guides/migrating-to-v2-api
<img width="798" alt="image" src="https://user-images.githubusercontent.com/490574/230113379-784c3b2e-9ad8-4db8-a8a9-7c4c629e1bff.png">

> Docs say I should use store instead of initialValues.

Maybe we should say "use useHydrateAtoms instead of ...". Can you send a PR?

**Reply by AlexTechNoir** [reply]
Wow, it works! Thank you! (I thought `useHydrateAtoms` need to be used only when we're fetching data in Next.js, I just had a local values in my atoms, but okay...)

Here's what I changed:
```javascript
import { Provider } from 'jotai'
import { someAtom, someOtherAtom } from '../store'

export default function MyApp({ Component, pageProps }) {
  const { initialState } = pageProps

  const HydrateAtoms = ({ someAtom, someOtherAtom, initialState, children }) => {
    useHydrateAtoms(initialState ? [ someAtom, someOtherAtom, initialState ] : [])
    return children
  }
  
  return(
    <Provider>
      <HydrateAtoms initialValues={initialState && [[ cartAtom, cartLengthAtom, initialState ]]}>
        <Component {...pageProps} />
      </HydrateAtoms>
    </Provider>
  )
}
```

> Can you send a PR?

I see changes in migration docs have already been made with that proper example. I initially followed a link in "releases" section and it points on old version 2.0.0 with `store`, whereas fresh docs version is 2.0.3 and is already with `useHydrateAtoms`
